contracts/v4/OmsPolicyV2.sol

```diff
@@ -0,0 +1,418 @@
+pragma solidity 0.4.24;
+
+import "./library/SafeMath.sol";
+import "./library/SafeMathInt.sol";
+import "./interface/IERC20.sol";
+import "./interface/IOracle.sol";
+import "./interface/UInt256Lib.sol";
+import "./common/Initializable.sol";
+import "./common/Ownable.sol";
+import "./Oms.sol";
+
+/**
+ * @title uOms Monetary Supply Policy
+ * @dev This is an implementation of the uOms Ideal Money protocol.
+ *      uOms operates symmetrically on expansion and contraction. It will both split and
+ *      combine coins to maintain a stable unit price.
+ *
+ *      This component regulates the token supply of the uOms ERC20 token in response to
+ *      market oracles.
+ */
+contract OmsPolicy is Ownable {
+    using SafeMath for uint256;
+    using SafeMathInt for int256;
+    using UInt256Lib for uint256;
+
+    event LogRebase(
+        uint256 indexed epoch,
+        uint256 exchangeRate,
+        int256 requestedSupplyAdjustment,
+        uint256 timestampSec
+    );
+
+    event LogTargetPrice(
+        uint256 lastTargetPrice,
+        uint256 newTargetPrice,
+        uint256 timestampSec
+    );
+
+    struct PriceLog {
```

Comment:
Might be a better design to move this PriceLog along with the calculateReferenceRate and *UnderFlow methods to the OraclePrice contract. To keep the 2 concerns separated. 

The Policy understands what to do with the target rate and the OraclePrice understands how to update the target rate. 

This way. In the future, if we add other currencies, we only need to update the OraclePrice contract.

Reply:
Not done yet. Raised an issue to track this.

Reply:
Moved calculateReferenceRate and relevant code to OraclePrice now. As it is not the concern of the Policy.